Week-1.cpp

```diff
@@ -0,0 +1,78 @@
+/*
+Problem: Longest Substring Without Repeating Characters
+Given a string, find the length of the longest substring without repeating characters.
+Examples:
+Given "abcabcbb", the answer is "abc", which the length is 3.
+Given "bbbbb", the answer is "b", with the length of 1.
+Given "pwwkew", the answer is "wke", with the length of 3. Note that the answer must be a substring, "pwke" is a subsequence and not a substring.
+*/
+#include <bits/stdc++.h>
+#include <string.h>
+using namespace std;
+
+int longestSubstr(string s) {
+	int len = s.length();
+	string space = " ";
+	
+	if (len==0) return 0;
+	
+	if (s.find(space)!=string::npos) {
+		return -1;
+	}
+	
+	int max_len=1, temp_len=1, prev_visit;
+	int last_visit[256];
+
+	
+	for (int i=0;i<256;i++) {
+		last_visit[i]=-1;
+	}
+	last_visit[s[0]]=0;
```

Comment:
You basically have the same exact line on line 43, except here it is hard coded to 0. That could mean this could be better organized so you only see it once, perhaps by moving line 43 to the beginning of the for loop and having the loop start at 0.

Reply:
Only thing I would possibly change is the space variable, typically if a variable is only used once I would just get rid of it and hard code the string, or make it into a const, but I don't remember how that works in C++ so that's fine in this case :)